### PR TITLE
Add better crypto error checking under compat flag

### PR DESCRIPTION
**Pull request by fhanau**
This adds lots of `CompatibilityFlags::Reader()` function parameters, unfortunately I am not aware of a better way to pass compat flags to a function except for using `IoContext::current()` which can't be used here.
Feel free to take a closer look at `validateRsaParams()`, it has several changes that are easy to get wrong. Other things to note are that key sizes that would lead to truncation by boringssl will be rejected and that the ECDH flags check has been moved from where the warning was.

### CODE REVIEW

**Comment by fhanau** [comment]
> See #742 for an attempt at simplifying how getting the compat flags would work. If we can get that landed then this PR can likely be simplified.

Certainly, yeah. I just rebased the PR since that will be needed anyway but I'll take a look at the compat flag PR later today.

File: src/workerd/api/crypto-impl-asymmetric-test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as assert from 'node:assert';
+let subtle = self.crypto.subtle;
```
Comment (jasnell) [inline src/workerd/api/crypto-impl-asymmetric-test.js:2]: `crypto.subtle` should be a global already. There shouldn't be a reason to assign to `subtle` here?

File: src/workerd/api/crypto-impl-asymmetric.c++ [hunk 2]
```diff
@@ -1010,18 +1004,22 @@ kj::OneOf<jsg::Ref<CryptoKey>, CryptoKeyPair> CryptoKey::Impl::generateRsa(
...
-  validateRsaParams(modulusLength, publicExponent.asPtr());
+  validateRsaParams(js, modulusLength, publicExponent.asPtr());
+  // boringssl silently uses (modulusLength & ~127) for the key size, i.e. it rounds down to the
+  // closest multiple of 128 bits. This can easily cause confusion when non-standard key sizes are
+  // requested.
+  // The `modulusLength` field of the resulting CryptoKey will be incorrect when the compat flag
+  // is disabled and the key size is rounded down, but since it is not currently used this is
+  // acceptable.
+  if (FeatureFlags::get(js).getStrictCrypto()) {
+    JSG_REQUIRE(!(modulusLength & 127), DOMOperationError,
```
Comment (jasnell) [inline src/workerd/api/crypto-impl-asymmetric.c++:1015]: Here too... can the Feature flag check be included in the `JSG_REQUIRE` condition?

**Review by jasnell — APPROVED** [review]
LGTM. Since I constantly forget to do this myself let's be sure to also have a cloudflare-docs PR opened that documents the new compat flag.

**Comment by fhanau** [comment]
> LGTM. Since I constantly forget to do this myself let's be sure to also have a cloudflare-docs PR opened that documents the new compat flag.

Added this just yesterday evening, would have mentioned it otherwise https://github.com/cloudflare/cloudflare-docs/pull/9388

**Comment by fhanau** [comment]
Incorporated your suggestions and did some slight refactoring. For the `JSG_REQUIRE()` conditions there's a lot of ways to define the condition parameter, I structured them in a way that I find most readable, but let me know if you have any more suggestions.

**Review by jasnell — APPROVED** [review]
Still LGTM!

**Pull request merged**